src/metrics/metrics.ts

```diff
@@ -65,79 +66,81 @@ export function countGamesInDataset(datasetPath: string): number {
-  basePieceSquares.set('c2', 'pc');
-  basePieceSquares.set('d2', 'pd');
-  basePieceSquares.set('e2', 'pe');
-  basePieceSquares.set('f2', 'pf');
-  basePieceSquares.set('g2', 'pg');
-  basePieceSquares.set('h2', 'ph');
-  basePieceSquares.set('a8', 'RA');
-  basePieceSquares.set('b8', 'NB');
-  basePieceSquares.set('c8', 'BC');
-  basePieceSquares.set('d8', 'Q');
-  basePieceSquares.set('e8', 'K');
-  basePieceSquares.set('f8', 'BF');
-  basePieceSquares.set('g8', 'NG');
-  basePieceSquares.set('h8', 'RH');
-  basePieceSquares.set('a7', 'PA');
-  basePieceSquares.set('b7', 'PB');
-  basePieceSquares.set('c7', 'PC');
-  basePieceSquares.set('d7', 'PD');
-  basePieceSquares.set('e7', 'PE');
-  basePieceSquares.set('f7', 'PF');
-  basePieceSquares.set('g7', 'PG');
-  basePieceSquares.set('h7', 'PH');
-
-  const chess = new Chess(); // Create a new instance of the Chess class
-  chess.loadPgn(game.moves);
-  const moveHistory = chess.history();
-
-  // duplicate the base map
-  const pieceSquares = new Map<Square, UnambiguousPieceSymbol>(
-    basePieceSquares
-  );
+  const chess = new Chess();
+  const moveGenerator = chess.historyGenerator(game.moves);
 
   // create an object to track distance value for each piece
   const distanceMap: { [key: string]: number } = {};
-  for (const piece of pieceSquares.values()) {
-    distanceMap[piece] = 0;
-  }
+
+  let lastMove: PrettyMove | null = null;
 
   // Initialize variables to keep track of the maximum distance and the piece
   let maxDistance = -1;
   let maxDistancePiece: UnambiguousPieceSymbol;
 
-  // TODO: we'll need to update the labels we use in cjsmin to be unique to do things this way
-  for (const { from, to } of moveHistory) {
-    // Calculate the file (column) distance by subtracting ASCII values
-    const fileDist = Math.abs(from.charCodeAt(0) - to.charCodeAt(0));
-    // Calculate the rank (row) distance by subtracting numeric values
-    const rankDist = Math.abs(Number(from[1]) - Number(to[1]));
-    // The distance moved is the maximum of fileDist and rankDist
-    const distance = Math.max(fileDist, rankDist);
-    // Get the piece that moved from the pieceSquares map
-    const movedPiece = pieceSquares.get(from);
-
-    // we'll update the map as pieces move. To avoid additional operations we only update (no delete). Can change this if we use this elsewhere
-    pieceSquares.set(to, movedPiece);
-    distanceMap[movedPiece] += distance;
-
-    if (distanceMap[movedPiece] > maxDistance) {
-      maxDistance = distanceMap[movedPiece];
-      maxDistancePiece = movedPiece;
+  // evaluate each move, update the correct unambiguous piece's distance
+  for (let moveInfo of moveGenerator) {
+    const { move, board } = moveInfo;
+
+    lastMove = move;
+    let piece = lastMove.unambiguousSymbol;
+
+    if (!distanceMap[piece]) {
+      distanceMap[piece] = 0;
+    }
+
+    const fromMove = moveInfo.move.from
+    const toMove = moveInfo.move.to
+
+    let distance = 0;
```

Comment:
pretty confusing bc this isn't used until the else block. Let's move it there and take out the redundant reassigning

Reply:
what specifically? all these lines?